src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -331,7 +331,22 @@ SHAMapStoreImp::run()
     ledgerDb_ = &app_.getLedgerDB();
 
     if (advisoryDelete_)
-        canDelete_ = state_db_.getCanDelete ();
+    {
+        canDelete_ = state_db_.getCanDelete();
+
+        // On startup, don't acquire ledgers from the network that we know
+        // can be deleted.
+        if (state_db_.getState().lastRotated)
+            minimumOnline_ = canDelete_.load() + 1;
+    }
+    if (!minimumOnline_)
+    {
+        // Or else don't acquire ledgers from the network older than
+        // the earliest persisted, if any.
+        auto const minSql = app_.getLedgerMaster().minSqlSeq();
+        if (minSql.has_value())
+            minimumOnline_ = *minSql;
+    }
```

Comment:
Good catch--actually, in that case, the desired behavior is to not try to acquire ledgers from the network based on this value. I fixed this in the minimumOnline() function. It would be nice to have a std::atomic<boost::optional<int>> but oh well.

Basically, if we don't know the specific lower bound of things to keep online, don't try to acquire historical ledgers based on this unknown.